src/test.js

```diff
@@ -537,18 +544,31 @@ function checkPollution() {
 
 // Will be exposed as QUnit.asyncTest
 function asyncTest( testName, expected, callback ) {
+	var warning;
+
+	if ( global.console && global.console.warn ) {
+		warning = "DEPRECATED: QUnit.asyncTest() will be removed in QUnit 2.0.\n" +
+			"Reference: https://api.qunitjs.com/QUnit.asyncTest/";
+		global.console.warn(warning);
+	}
+
 	if ( arguments.length === 2 ) {
 		callback = expected;
 		expected = null;
 	}
 
-	QUnit.test( testName, expected, callback, true );
+	QUnit.test( testName, expected, callback, warning, true );
 }
 
 // Will be exposed as QUnit.test
-function test( testName, expected, callback, async ) {
+function test( testName, expected, callback, warning, async ) {
 	if ( focused )  { return; }
 
+	if ( this === global ) {
+		warning = "DEPRECATED: Global 'test()' will be removed in QUnit 2.0, " +
+			"use 'QUnit.test() instead";
+	}
```

Comment:
this check is confusing, I also wonder what could happen if I need to copy the function at QUnit.test into a variable.